### PR TITLE
[v6r20] fix fts3 deadlock

### PR DESCRIPTION
**Pull request by chaen**
BEGINRELEASENOTES

*DMS

NEW:  FTS3 add kicking of stuck jobs
FIX: FTS3 update files in sequence to avoid mysql deadlock
CHANGE: Canceled is not a final state for FTS3 Files
CHANGE: FTS3Operations are finalized if the Request is in a final state (instead of Scheduled)

ENDRELEASENOTES

### CODE REVIEW

**Comment by chaen** [comment]
Yep, done ! Thanks.

File: DataManagementSystem/Agent/FTS3Agent.py [hunk 1]
```diff
@@ -1,5 +1,34 @@
...
+    PollingTime = 120
+    MaxThreads = 10
+    # How many Operation we will treat in one loop
+    OperationBulkSize = 20
+    # How many Job we will monitor in one loop
+    JobBulkSize = 20
+    # Max number of files to go in a single job
+    MaxFilesPerJob = 100
+    # Max number of attempt per file
+    maxAttemptsPerFile = 256
```
Comment (andresailer) [inline DataManagementSystem/Agent/FTS3Agent.py]: For consistency, could we start this option also with a capital letter?

File: docs/source/AdministratorGuide/Systems/DataManagement/fts3.rst [hunk 2]
```diff
@@ -75,16 +75,7 @@ FTS3Agent
...
-
-* `OperationBulkSize` (default 20): How many Operation we will treat in one loop
-* `JobBulkSize` (default 20): How many Job we will monitor in one loop
-* `MaxFilesPerJob` (default 100): Max number of files to go in a single job
-* `maxAttemptsPerFile` (default 256): Max number of attempt per file
-* `DeleteGraceDays` (default 180): days before removing jobs
-* `DeleteLimitPerCycle` (default 100): Max number of deletes per cycle
-* `KickAssignedHours ` (default 1): hours before kicking jobs with old assignment tag
-* `KickLimitPerCycle` (default 100):  Max number of kicks per cycle
+See :py:mod:`~DIRAC.Core.DataManagementSystem.Agents.FTS3Agent` for configuration details.
```
Comment (andresailer) [inline docs/source/AdministratorGuide/Systems/DataManagement/fts3.rst]: `~DIRAC.DataManagementSystem.Agent.FTS3Agent`

Comment (chaen) [reply docs/source/AdministratorGuide/Systems/DataManagement/fts3.rst]: Thanks !

File: DataManagementSystem/Agent/FTS3Agent.py (hunk 1 above)
Comment (chaen) [reply DataManagementSystem/Agent/FTS3Agent.py]: Absolutely

**Comment by atsareg** [comment]
There are conflicts in this branch. I guess both modifications should be kept. Please, check

**Comment by chaen** [comment]
Fixed

**Pull request merged**